### PR TITLE
feat: Setup E2E tests for the dsp-operator

### PR DESCRIPTION
**Pull request by harshad16**
Setup E2E tests for the dsp-operator

## Description
E2e test would allow openshift-ci to setup dsp-operator with help of opendatahub-operator and test a run of pipeline execution on the setup.

- Created the e2e tests based on the pattern on odh-manifests e2e tests.
- used the sample crd of dsp-operator from manifests.
- Include parts of [data-science-pipeline tests](https://github.com/opendatahub-io/data-science-pipelines/blob/master/tests/basictests/ds-pipelines.sh) for testing the pipelinerun.

## How Has This Been Tested?

with the help of makefile and openshift cluster, try the following:
```
cd tests
make build
make run
```

Tested on openshift cluster with version: 4.12.2

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

Note: Removed Monitoring for the first PR, will include it in following PR

### CODE REVIEW

File: tests/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ARG REPO=data-science-pipelines-operator
+ARG ODS_CI_REPO=https://github.com/red-hat-data-services/ods-ci
+ARG ODS_CI_GITREF=releases/1.7.0-5
+ARG OC_CLI_URL=https://mirror.openshift.com/pub/openshift-v4/amd64/clients/ocp/latest/openshift-client-linux.tar.gz
+
+ENV HOME /root
+WORKDIR /root
+
+RUN dnf install -y bc git go-toolset python3-pip unzip && \
+    git clone https://github.com/crobby/peak $HOME/peak && \
```
[the remark this answers is not in the document]
Comment (harshad16) [reply tests/Dockerfile:16]: ah right, good catch
replacing it with https://github.com/AICoE/peak

[the remark this answers is not in the document]
Comment (harshad16) [reply tests/Dockerfile:16]: The aicoe/peak and crobby/peak have a lot of differences, investigating the repercussion of changing this.

[the remark this answers is not in the document]
Comment (harshad16) [reply tests/Dockerfile:16]: There seems to be  multiple copies of peak 
https://github.com/AICoE/peak/network/members
https://github.com/AICoE/peak-fork/network/members

the one from crobby/peak is fork of peak-fork, they all have different changes
the peak from aicoe, is not giving smooth result. 
more work would be required to go further over this.
can we skip for now @DharmitD ?

[the remark this answers is not in the document]
Comment (DharmitD) [reply tests/Dockerfile:16]: Makes sense, let's skip this for now.

**Review by DharmitD — COMMENTED** [review]
/lgtm

File: tests/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (anishasthana) [reply tests/Dockerfile:16]: Could we create a fork of Chad's peak into opendatahub-io?

[the remark this answers is not in the document]
Comment (harshad16) [reply tests/Dockerfile:16]: I will check with OWNERs of opendatahub-io org and try to get this.
once that is done, will change the reference in future PR.

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+make build
+make run
+```
+
+## Cleaning up after your test run (optional)
+
+Only run the following if you want to eliminate your Open Data Hub installation.
+
+To cleanup the Open Data Hub installation after a test run, you can run `make clean`.
+Running `make clean` **will wipe your Open Data Hub installation** and delete the project.
```
[the remark this answers is not in the document]
Comment (harshad16) [reply tests/README.md:21]: The README was from the odh-manifests repo, i will try to address this overall everywehre.

**Pull request merged**